### ISSUE TITLE
Nomad drain_on_shutdown improvement

### ISSUE THREAD

**Issue by suikast42** [hashicorp/nomad#17093]
The new release of nomad v 1.5.4 intruduces the possibility to auto drain nodes via client config 

```hcl
  drain_on_shutdown {
    deadline           = "30s"
    force              = false
    ignore_system_jobs = false
  }
```

That works so far so good but have some sideeffects.

1. Everytime if nomad worker restarts over systemd then node is drained and ineligible. 
2. Evertime on first boot of the machine the node is ineligible

Systemd restarts are required for example when adding or removing a host volume ( when not using the csi driver )  or changing the config config.

### Suggestion
 
1. Adding a agent setting that makes the node eligable again in automated way 
2. Give the agent the possibility to ignore node drain ( for systemd restart for example)

**Comment by tgross** [comment]
@suikast42 for (2), by design the `drain_on_shutdown` is intended to interact with the options `leave_on_interrupt` and `leave_on_terminate` to handle the problems you're describing (ref https://developer.hashicorp.com/nomad/docs/configuration/client#drain_on_shutdown-block). You just need to configure `systemd` (and your configuration management system that might be rolling out config changes to host volumes) to send the appropriate signal for restarts with and without drains. For example, you might use systemd's default `SIGINT` for normal restarts without drains, but configure `leave_on_terminate` + `drain_on_shutdown` so that you can "turn down" a client when you're shutting down the whole host.

**Comment by suikast42** [comment]
There is a little gap in docu paage. 

It lists only the latest versions not the minors. 

The drain_on_shutdown  for example not exists in the 1.5.1 version 

> May 05 12:43:03 worker-01 nomad[19946]: ==> Error loading configuration from /etc/nomad.d: Error loading /etc/nomad.d/client.hcl: client unexpected keys drain_on_shutdown
May 05 12:43:03 worker-01 systemd[1]: nomad.service: Main process exited, code=exited, status=1/FAILURE

I am blocked to use higher version of nomad because of this bug https://github.com/hashicorp/nomad/issues/17079

But for clear understanding 

If I want that the node drain occurs only when shutdown or reboot the host then I have to set leave_on_terminate = true and leave_on_interrupt= false ?

**Comment by suikast42** [comment]
And how can I handle that the  node is eligable again after boot or restart ?  

I handle this currently with systemd poststart hook but is not ideal imho.

**Comment by tgross** [comment]
> There is a little gap in docu page.
>
> It lists only the latest versions not the minors.
> 
> The drain_on_shutdown for example not exists in the 1.5.1 version

Yeah that's an unfortunate bug in our "versioned docs" process. We're still working on figuring out how to make it so that doesn't happen in the future. But for now yes not available until 1.5.4 (but we'd recommend going to 1.5.5 and skipped 1.5.4 entirely).

> If I want that the node drain occurs only when shutdown or reboot the host then I have to set leave_on_terminate = true and leave_on_interrupt= false ?

Your systemd unit file determines what signal gets sent based on the systemd unit file's `KillSignal` directive. So if you're using the [`nomad.service`](https://github.com/hashicorp/nomad/blob/main/.release/linux/package/usr/lib/systemd/system/nomad.service#L18) file we ship in our official Linux packages, then you'd want to configure `leave_on_interrupt = true` and `leave_on_terminate = false`. And then if you wanted to restart Nomad without draining you'd use `systemctl kill -s SIGTERM nomad` instead.

> And how can I handle that the node is eligable again after boot or restart ?
>
> I handle this currently with systemd poststart hook but is not ideal imho.

Currently there's no built-in way to do that. So yeah you'd want to do it with a post start hook or the like for now. But I'll leave this issue open as a feature request to look at switching eligibility.

**Comment by suikast42** [comment]
Here is my workrroud for that:

Maybe thereis a better way or that can help someone other.

1. in nomad.service for agent 
```systemd
ExecStartPost=systemctl restart nomad.eligtion.service
```
2. nomad.eligtion.service
```systemd
[Unit]
Description=Nomad eligtion service. Wait until nomad is up and make it eligible again
Documentation=https://www.nomadproject.io/docs/


[Service]
Environment=NOMAD_ADDR=https://localhost:4646
Environment=NOMAD_CACERT=/usr/local/share/ca-certificates/cloudlocal/cluster-ca-bundle.pem
Environment=NOMAD_CLIENT_CERT=/etc/opt/certs/nomad/nomad.pem
Environment=NOMAD_CLIENT_KEY=/etc/opt/certs/nomad/nomad-key.pem

ExecStart=/bin/bash -c 'if [ ! -f "/home/user/noteligt" ] ; then  (while ! nomad  node eligibility  -enable -self  >/dev/null; do echo "Waiting for port nomad agent runs..."; sleep 1; done); sleep 1;fi'

[Install]
WantedBy=multi-user.target
```

you can skip the auto eligtion by touch a file /home/user/noteligt before node shutdown
